hacspec/src/vec_integers.rs

```diff
@@ -21,15 +21,27 @@ macro_rules! _implement_numeric_unsigned_public {
         }
         impl Eq for $name {}
 
+        impl $name {
+            /// Check if the two sequences are compatible, i.e. have the same
+            /// length.
+            fn compatible(&self, other: &Self) -> bool {
+                debug_assert!(
+                    self.len() == other.len(),
+                    "Can't combine two sequences that don't have the same length."
+                );
+                if self.len() != other.len() {
+                    return false;
+                }
+                return true;
+            }
+        }
+
         /// **Warning**: wraps on overflow.
         impl Add for $name {
             type Output = $name;
             #[cfg_attr(feature="use_attributes", library(hacspec))]
             fn add(self, rhs: $name) -> $name {
-                debug_assert!(
-                    self.len() == rhs.len(),
-                    "Can't add two sequences that don't have the same length."
-                );
+                self.compatible(&rhs);
```

Comment:
So `self.compatible` yields only a `debug_assert`, so in a release build, `+` would not panic on sequences that don't have the same length. What is the semantic of addition on sequences that don't have the same length ? If we don't want to give such a semantic, we should put an `assert` instead of a `debug_assert` inside `self.compatible`.

Reply:
Ha, good catch! Will do.